aws/resource_aws_athena_database.go

```diff
@@ -38,18 +38,73 @@ func resourceAwsAthenaDatabase() *schema.Resource {
 				Optional: true,
 				Default:  false,
 			},
+			"encryption_key": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
+func getResultConfig(d *schema.ResourceData) (*athena.ResultConfiguration, error) {
+	resultConfig := athena.ResultConfiguration{
+		OutputLocation: aws.String("s3://" + d.Get("bucket").(string)),
+	}
+
+	e := d.Get("encryption_key").([]interface{})
+	if len(e) <= 0 {
+		return &resultConfig, nil
+	}
+
+	data := e[0].(map[string]interface{})
+	keyType := data["type"].(string)
+	keyID := data["id"].(string)
+
+	if len(keyType) <= 0 {
```

Comment:
Since the schema already has this set as `Required: true` this error handling is extraneous.